### ISSUE TITLE
File submission

### ISSUE THREAD

**Issue by OhhTuRnz** [mit-ll/spacegym-kspdg#12]
Hi,

While trying to submit my own results, I found that an error didn't let me submit them. I figured out that the name that i was using for the agent used a whitespace and the evaluation makes some sort of hash that will detect if i change any data from the results. How do I proceed? Do i have to try to get the same results from my agent?

Thanks in advance.

**Comment by rallen10** [comment]
Correct, you cannot modify the results txt file in any way, even if it is just removing a whitespace. In principle, if you just add back whatever whitespace you removed from the agent name, it should be accepted; however, whitespace can be tricky and I don't know exactly how the hashing function will respond if it happens to identify it as a different whitespace. 

If you can't get your results file back to the exact state it was in when generated, then you will need to re-run your experiment and take whatever new results come out.

**Comment by OhhTuRnz** [comment]
I removed the whitespace from my agents' name and uploaded a test, but this error still happens:

![image](https://github.com/mit-ll/spacegym-kspdg/assets/72455516/2e66d592-231c-4fe4-b791-04e9268f1d07)

My user id and user key are apparently ok, i checked it out with my team. My colleague could manage to upload his files but i can't get in touch right now so I'd appreciate if you know what is actually happening

**Comment by rallen10** [comment]
Is this the original error you were receiving or is this a new error? This error should only happen if your `user_id` does not exactly match `"MIT-ARCLab"`. Can you confirm that your eval config yaml file has a line that is identical to the following:

```
user_id: "MIT-ARCLab"
```

There can be no whitespaces in the "MIT-ARCLab"

**Comment by OhhTuRnz** [comment]
[kspdg_results_20231126_134519 (copy).txt](https://github.com/mit-ll/spacegym-kspdg/files/13537597/kspdg_results_20231126_134519.copy.txt)

I'm attaching one example removing the user-key

**Comment by OhhTuRnz** [comment]
[kspdg_results_20231126_134519 (copy).txt](https://github.com/mit-ll/spacegym-kspdg/files/13537605/kspdg_results_20231126_134519.copy.txt)

I'm attaching one submission file removing sensitive info

**Comment by rallen10** [comment]
hmmm...Yeah let me debug this. I can't do it at this exact moment; might be a few hours

**Comment by OhhTuRnz** [comment]
And yes, the only two errors I got were this and checksum (for the hash I suppose)

**Comment by OhhTuRnz** [comment]
> hmmm...Yeah let me debug this. I can't do it at this exact moment; might be a few hours

Thanks!

**Comment by rallen10** [comment]
I've confirmed that I can reproduce the error with your user_id. However I also see that results actually seemed to be logged after the error is thrown, which doesn't make any sense to me yet. Can you confirm that the score of the agent you tried to upload for `PE1_E3_I1_V1` was `251.251`?

It looks like I'll need to debug this further, but for the moment, it actually looks like your score is logged

**Comment by rallen10** [comment]
So here is what I've found so far:

- When debugging locally on my machine, I do not get the user_id error
- When I try uploading results for MIT-ARCLab on the server, I do get the error, BUT...
- The results upload anyway! Look up the SB1_E1_I5_V1 results on [https://edges.ll.mit.edu](https://edges.ll.mit.edu) and you will see an agent called `ADMIN-DEBUG-PASSIVE` that I successfully uploaded in spite of seeing the same user_id error

Therefore my suspicion is that the error message is some kind of artifact from prior upload attempts that hasn't been cleaned out, but that current attempts are actually getting uploaded.

I need to fix this issue, as well as the black entries, but you should check if you can actually upload results in spite of the error message appearing

**Comment by OhhTuRnz** [comment]
Ok, I will try to upload them. It may be a conflict with my environment actually. Thanks

**Comment by rallen10** [comment]
We've implemented a fix on the server for this issue. Let me know if you run into this again. In the meantime I will close this issue

**Comment by OhhTuRnz** [comment]
I've ran into this again. My agent passed SB1 problem and I wanted to upload it, I checked if the submission was successful even if the error appeared, but i can't see it in the leaderboard. May i send you the file via private message?

Thank you in advance

**Comment by rallen10** [comment]
yes, please send the file to me
